### PR TITLE
create-replacement-pr: approve and merge applicable PRs

### PR DESCRIPTION
**Pull request by carlocab**
Let's automatically approve replacement PRs where the original PR was
approved, and enable auto-merge on replacement PRs where we've already
uploaded bottles for.

As in our other workflows, we need to use `GITHUB_TOKEN` to approve
(because the replacement PR is created by @BrewTestBot) and
`HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN` to enable auto-merge (so that our
merge queue CI runs).

### CODE REVIEW

**Review by ZhongRuoyu — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]
I'm cautiously fine with this but: I'm wary of automatically approving PRs unless we're 100% sure there's literally no possible way you can get a change in here that's not been already reviewed by a human.

I still think in the best case scenario: every green PR needs reviewed once by a one human before merge.

**Pull request merged**
